### PR TITLE
Improve installation instructions

### PR DESCRIPTION
**Pull request by waldyrious**
- Various wording, punctuation and formatting tweaks
- Add a separate section for Homebrew, since it works for both macOS and Linux
- Break some long lines, especially those with links, to allow for cleaner diffs

### CODE REVIEW

**Comment by sebastian-xyz** [comment]
Could you also update those `--all-features` flags for building it locally? They should be `--features=stable`
I wanted to do this for a while but somehow forgot it😅

**Comment by waldyrious** [comment]
Sure :) will do.

File: installation.md [hunk 1]
```diff
@@ -91,34 +105,37 @@ For Rust to work properly, you'll need to have a compatible compiler suite insta
...
 cargo install nu
 ```
 
-You can also install Nu with all the bells and whistles (be sure to have installed the [dependencies](https://book.nushell.sh/en/installation#dependencies) for your platform):
+You can also install Nu with all the bells and whistles (be sure to have installed the
+[dependencies](https://book.nushell.sh/en/installation#dependencies) for your platform):
 
 ```
-cargo install nu --all-features
+cargo install nu --features=stable
```
Comment (waldyrious) [inline installation.md]: @sebastian-xyz The [README](https://github.com/nushell/nushell#local) contains an almost identical passage to this one, but also includes the `--all` flag:

> You can also build Nu yourself with all the bells and whistles (be sure to have installed the dependencies for your platform), once you have checked out this repo with git:
>
>     cargo build --all --features=stable

Which is the correct command line?

Comment (sebastian-xyz) [reply installation.md]: I'm not sure. According to the blog posts it is `--features=stable`. But I think `--all --features=stable` comes with more plugins.

@jonathandturner can you help us out?

Comment (jonathandturner) [reply installation.md]: Yup, you got it. Building locally `--all --features=stable` gives you the most features and plugins

Comment (waldyrious) [reply installation.md]: Ok, thanks! I've added `--all` to the commands.

**Comment by sebastian-xyz** [comment]
Thank you :)

**Pull request merged**